lib/improver/tests/ensemble_calibration/ensemble_calibration/test_EstimateCoefficientsForEnsembleCalibration.py

```diff
@@ -589,33 +640,61 @@ def test_realizations_predictor_estimate_coefficients(self):
             predictor_of_mean_flag,
             estimate_coefficients_from_linear_model_flag,
             no_of_realizations=self.no_of_realizations)
-        self.assertArrayAlmostEqual(result, data)
+        self.assertArrayAlmostEqual(
+            self.expected_realizations_predictor_with_linear_model, result)
 
     @ManageWarnings(
         ignored_messages=IGNORED_MESSAGES, warning_types=WARNING_TYPES)
-    def test_mean_predictor_estimate_coefficients_nans(self):
+    def test_mean_predictor_estimate_coefficients_masked_halo(self):
         """
         Test that the plugin returns the expected values for the initial guess
         for the calibration coefficients, when the ensemble mean is used
-        as the predictor, when one value from the input data is set to NaN.
-        The coefficients are estimated using a linear model,
+        as the predictor. The coefficients are estimated using a linear model,
         where there is an offset of one between the truth and the forecast
         during the training period. Therefore, in this case the result of the
-        linear regression is a gradient of 1 and an intercept of 1.
+        linear regression is a gradient of 1 and an intercept of 1. In this
+        case the original data has been surrounded by a halo of masked nans,
+        which gives the same coefficients as the original data.
         """
-        data = np.array([0., 1., 1., 1.], dtype=np.float32)
         estimate_coefficients_from_linear_model_flag = True
 
-        self.current_forecast_predictor_mean.data = (
-            self.current_forecast_predictor_mean.data.filled())
-        self.current_forecast_predictor_mean.data[0][0] = np.nan
-
         plugin = Plugin(self.distribution, self.desired_units)
         result = plugin.compute_initial_guess(
-            self.truth, self.current_forecast_predictor_mean,
+            self.truth_masked_halo,
+            self.current_forecast_predictor_mean_masked_halo,
             self.predictor_of_mean_flag,
             estimate_coefficients_from_linear_model_flag)
-        self.assertArrayAlmostEqual(result, data)
+
+        self.assertArrayAlmostEqual(
+            self.expected_mean_predictor_with_linear_model, result)
+
```

Comment:
I would like a test with nans in the data, not because they are sea points, but just because they are invalid data points. These used to be masked out, but I don't they they are anymore, so I'd like to know whether they make EMOS behave oddly/error/etc.